x-pack/plugins/maps/public/ems_autosuggest/ems_autosuggest.test.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { suggestEMSTermJoinConfig } from './ems_autosuggest';
+import { FORMAT_TYPE } from '../../common';
+import { FeatureCollection } from 'geojson';
+
+class MockFileLayer {
+  private readonly _url: string;
+  private readonly _id: string;
+  private readonly _fields: Array<{ id: string }>;
+
+  constructor(url: string, fields: Array<{ id: string }>) {
+    this._url = url;
+    this._id = url;
+    this._fields = fields;
+  }
+  getDefaultFormatUrl() {
+    return this._url;
+  }
+
+  getFields() {
+    return this._fields;
+  }
+
+  getDefaultFormatType() {
+    return FORMAT_TYPE.GEOJSON;
+  }
+
+  hasId(id: string) {
+    return id === this._id;
+  }
+}
+
+jest.mock('../util', () => {
+  return {
+    async getEmsFileLayers() {
+      return [
+        new MockFileLayer('world_countries', [{ id: 'iso2' }, { id: 'iso3' }]),
+        new MockFileLayer('zips', [{ id: 'zip' }]),
+      ];
+    },
+    async fetchGeoJson(url: string): Promise<FeatureCollection> {
+      if (url === 'world_countries') {
+        return ({
+          type: 'FeatureCollection',
+          features: [
+            { properties: { iso2: 'CA', iso3: 'CAN' } },
+            { properties: { iso2: 'US', iso3: 'USA' } },
+          ],
+        } as unknown) as FeatureCollection;
+      } else if (url === 'zips') {
+        return ({
+          type: 'FeatureCollection',
+          features: [{ properties: { zip: '40204' } }, { properties: { zip: '40205' } }],
+        } as unknown) as FeatureCollection;
+      } else {
+        throw new Error(`unrecognized mock url ${url}`);
+      }
+    },
+  };
+});
+
+describe('suggestEMSTermJoinConfig', () => {
+  test('no info provided', async () => {
+    const termJoinConfig = await suggestEMSTermJoinConfig({});
+    expect(termJoinConfig).toBe(null);
+  });
+
+  describe('validate common column names', () => {
+    [
```

Comment:
nit: I always find iterating over test stubs like this harder to follow then just breaking each out into its own test case. There is not very much duplicated code and it makes the asserts easier to see when everything is just inlined in a single location. I would recommend breaking into individual test calls.